server/sonar-server/src/main/java/org/sonar/server/ws/DeprecatedRestWebServiceFilter.java

```diff
@@ -117,4 +110,88 @@ protected String readParam(String key) {
     }
   }
 
+  private static class Response {
+    private final HttpServletRequest request;
+    private final Map<String, String> additionalParams = new HashMap<>();
+    private final String originalPath;
+    private String redirectedPath;
+
+    Response(HttpServletRequest request) {
+      this.request = request;
+      this.originalPath = request.getRequestURI().replaceFirst(request.getContextPath(), "");
+      init();
+    }
+
+    void init() {
+      String method = request.getMethod();
+      Optional<String> key = getKey();
+      Optional<String> value = getValue();
+      Optional<String> component = getComponent();
+      switch (method) {
+        case "POST":
+        case "PUT":
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 9 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1151)